drivers/led/lp5569.c

```diff
@@ -77,6 +77,23 @@ static inline int lp5569_led_off(const struct device *dev, uint32_t led)
 	return lp5569_led_set_brightness(dev, led, 0);
 }
 
+static int lp5569_write_channels(const struct device *dev, uint32_t start_channel,
+				 uint32_t num_channels, const uint8_t *buf)
+{
+	const struct lp5569_config *config = dev->config;
+	uint32_t i2c_len = num_channels + 1;
+	uint8_t i2c_msg[i2c_len];
```

Comment:
Please avoid such uncontrolled / dynamic allocations on the stack. It makes me really nervous :) At the very least you need to limit the buffer size...